synapse/handlers/room_member.py

```diff
@@ -710,17 +777,98 @@ def _lookup_3pid(self, id_server, medium, address):
                 return data["mxid"]
 
         except IOError as e:
-            logger.warn("Error from identity server lookup: %s" % (e,))
+            logger.warn("Error from v1 identity server lookup: %s" % (e,))
+
+        return None
+
+    @defer.inlineCallbacks
+    def _lookup_3pid_v2(
+        self, id_server, id_access_token, medium, address, hash_details
+    ):
+        """Looks up a 3pid in the passed identity server using v2 lookup.
+
+        Args:
+            id_server (str): The server name (including port, if required)
+                of the identity server to use.
+            id_access_token (str): The access token to authenticate to the identity server with
```

Comment:
In *theory* the hash endpoint checks that you gave an access token. In *practice* it might not. I think you should figure out how to handle that case.

Reply:
I've changed around `_lookup_3pid` to only call `_lookup_3pid_v2` if it gets an `id_access_token`, which should solve this now.

Indeed the original code was not backwards compatible and things would fail if an `id_access_token` parameter was not supplied. That should now be fixed.